client/ctclient/ctclient.go

```diff
@@ -342,29 +345,47 @@ func main() {
 		}
 		opts.PublicKey = string(pubkey)
 	}
-	logClient, err := client.New(*logURI, httpClient, opts)
+
+	var err error
+	var logClient *client.LogClient
+	var checkClient client.CheckLogClient
+	if *dnsBase != "" {
```

Comment:
Could we make checkClient a logClient that rejects the stuff it can't do? Then this initialization and the checks below would be simpler.

Reply:
I think it's better to catch things at compile-time than at run-time whenever possible, so I'm not keen on the `unimplemented` approach.  (But then, I've been known to write Ocaml for fun...)

Reply:
This is caught at runtime though? If you set options for DNS and something it doesn't support.

Reply:
Maybe I misunderstood - I thought you were suggesting making the `DNSClient` implement the full `LogClient` interface, then removing the operation-specific checks and letting the `errUnimplemented` flow out to the user.

Reply:
Yes, that's what I meant. It doesn't seem that different and removes extra logic.